lib/nexpose/wait.rb

```diff
@@ -0,0 +1,127 @@
+module Nexpose
+
+  class Wait
+    attr_reader :error_message, :ready, :retry_count, :timeout, :polling_interval
+
+
+    def initialize(retry_count: nil, timeout: nil, polling_interval: nil)
+      @error_message = "Default General Failure in Nexpose::Wait"
+      @ready = false
+      @retry_count = retry_count.to_i
+      @timeout = timeout
+      @polling_interval = polling_interval
+    end
+
+
+    def ready?
+      @ready
+    end
+
+
+    def for_report(nexpose_connection: nil, report_id: nil)
+      poller = Nexpose::Poller.new(timeout: @timeout, polling_interval: @polling_interval)
+      poller.wait(report_status_proc(nexpose_connection: nsc, report_id: report_id))
+      @ready = true
+      rescue TimeoutError
+        retry if timeout_retry?
+        @error_message = "Timeout Waiting for Report to Generate - Report Config ID: #{report_id}"
+      rescue NoMethodError => error
+        @error_message = "Error Report Config ID: #{report_id} :: Report Probably Does Not Exist :: #{error}"
+      rescue => error
+        @error_message = "Error Report Config ID: #{report_id} :: #{error}"
+    end
+
+
+    def for_integration(nexpose_connection: nil, scan_id: nil, status: 'finished')
+      poller = Nexpose::Poller.new(timeout: @timeout, polling_interval: @polling_interval)
+      poller.wait(integration_status_proc(nexpose_connection: nsc, scan_id: scan_id, status: status))
+      @ready = true
+      rescue TimeoutError
+        retry if timeout_retry?
+        @error_message = "Timeout Waiting for Integration Status of '#{status}' - Scan ID: #{scan_id}"
+      rescue Nexpose::APIError => error
+        @error_message = "API Error Waiting for Integration Scan ID: #{scan_id} :: #{error.req.error}"
+    end
+
+
+    def for_judgment(proc: nil, desc: nil)
+      poller = Nexpose::Poller.new(timeout: @timeout, polling_interval: @polling_interval)
+      poller.wait(proc)
+      @ready = true
+      rescue TimeoutError
+        retry if timeout_retry?
+        @error_message = "Timeout Waiting for Judgment to Judge. #{desc}"
+    end
+
+
+    private
+
+
+      def report_status_proc(nexpose_connection: nil, report_id: nil)
+        Proc.new { nsc.last_report(report_id).status == 'Generated' }
+      end
+
+
+      def integration_status_proc(nexpose_connection: nil, scan_id: scan_id, status: status)
```

Comment:
Unused method argument - `nexpose_connection`.<br>Line is too long. [92/80]<br>Inconsistent indentation detected.